src/frame.js

```diff
@@ -108,15 +108,18 @@ function frame(imgBuff, ts) {
 	let _histogram = null;
 	let _progress = null;
 	let _perceptualProgress = null;
+	let _parsedImage = null;
 
 	return {
 		getHistogram: function () {
 			if (_histogram) {
 				return _histogram;
 			}
 
-			const pixels = jpeg.decode(imgBuff);
-			_histogram = convertPixelsToHistogram(pixels);
+			if (!_parsedImage) {
```

Comment:
sounds good. everything else was using internal state only, no `this.*`, so wasn't sure if that was on purpose. Tests look good :)